src/Main.hs

```diff
@@ -84,12 +84,18 @@ minWithEmpty :: [Double] -> Double
 minWithEmpty [] = -1
 minWithEmpty list = minimum list
 
-trace x y 
-    | isNothing intersection = white
-    | otherwise = getColorFromIntersection intersection
-    where intersection = closestIntersection (generateRay camera sceneWidth sceneHeight x y) objects
+trace :: Int -> Int -> PixelRGB8
+trace x y =
+    let
+      ray =  generateRay camera sceneWidth sceneHeight x y
+      intersections = closestIntersection ray objects
+    in
+      case intersections of Nothing -> white
+                            Just distObj -> getColorFromIntersection distObj
+
 
-getColorFromIntersection (Just ( _ , Object _ (Material color))) = color
+getColorFromIntersection :: (Double, Object) -> PixelRGB8
+getColorFromIntersection ( _ , Object _ (Material color)) = color
```

Comment:
yep, exactly. I generally feel than unless you're function is doing something specific to the Maybe, then the function shouldn't be unwrapping that itself. instead you should use that function to map or flatmap a maybe, and do the unwrapping higher up